lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt

```diff
@@ -889,6 +889,49 @@ internal class EvaluatingCompiler(
         }
     }
 
+    private fun compileLimit(limitClause: ExprNode, env: Environment) : Long {
```

Comment:
In general, any `compile*` function should return a thunk.  This appears to be compiling and evaluating.  Compilation and evaluation should be separate steps.

Reply:
In the new commit, I separated out the compilation ([L955](https://github.com/partiql/partiql-lang-kotlin/blob/ad2aa3e73f549c3ff66605d254c2c5b7dfdbc277/lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt#L955)) from the evaluation ([L892](https://github.com/partiql/partiql-lang-kotlin/blob/ad2aa3e73f549c3ff66605d254c2c5b7dfdbc277/lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt#L892)).